src/main/kotlin/com/manimdsl/runtime/ASTExecutor.kt

```diff
@@ -667,6 +680,57 @@ class VirtualMachine(
             )
         }
 
+
+        private fun executeWhileStatement(whileStatementNode: WhileStatementNode): ExecValue {
+            if (showMoveToLine) addSleep(0.5)
+
+            var conditionValue: ExecValue
+            var execValue: ExecValue
+
+
+            while (true) {
```

Comment:
Just wondering whether we can avoid a while true here? It does seem hard though because you have to check for runtime errors and then check the bool value.

Anyone else have any ideas for this?

Reply:
@rbanerjee20  you can just set the pc to be the condition line or after the loop depending on if the condition is true or false

Reply:
Would we not need to recursively call `executeWhileStatement` in that case to get the frame to be run several times? (as many times as needed) @adityagoel4512

Reply:
Also, I was looking into this, and if the user has an infinite loop in their program, it should just run forever (potentially using `while true` internally?). It's the Halting Problem 😅

Reply:
Should we just keep a counter like we do with frame depth, and if the while loop goes on for more than 1000 times we throw a runtime error?

Reply:
Yeah that sounds like a good plan. In practical terms, it’s an animation, so we can just throw a runtime error if we think it’s going on for too long

Reply:
I’ll add it in

Reply:
> Would we not need to recursively call `executeWhileStatement` in that case to get the frame to be run several times? (as many times as needed) @adityagoel4512

nope. if you move back up to the condition line the frame will take care of the rest - it should just reexcute as before but with the updated state of variables (which is what we want in a loop)

Reply:
Like with assembly we jmp to a label here we do a move to line and should be good

Reply:
I think the issue is that this function's return type is `ExecValue` but it doesn't always return - it shouldn't return anything when starting loop execution again or when it reaches a continue statement. Not really sure how to get around that

Reply:
I don't think there would be a problem with returning something so long as the pc has been set to the correct location as executeStatement just would reexecute the line. This is OK however and I took a while to get back to you so feel free to merge

Reply:
the condition line that is